app/models/project.server.ts

```diff
@@ -771,3 +774,75 @@ export async function deleteProject(projectId: string, isAdmin: boolean) {
   }
   return true;
 }
+
+export async function archiveProject(
+  projectId: string,
+  profileId: string,
+  isAdmin: boolean
+) {
+  const currentProject = await db.projects.findUniqueOrThrow({
+    where: { id: projectId },
+    select: {
+      ownerId: true,
+    },
+  });
+
+  const projectMembers = await db.projectMembers.findMany({
+    where: { projectId },
+    select: {
+      profileId: true,
+    },
+  });
+
+  if (!isAdmin)
+    validateIsTeamMember(profileId, projectMembers, currentProject.ownerId);
```

Comment:
I'm thinking we should move these validation functions out of the *.server.ts files, and instead do them inside the `action` functions. That way the database layer will be cleaner, and authorization logic will live close to the view/controller layer.

What do you think?

Reply:
For an example of this, look at the `app/routes/projects/$projectId/updateRelatedProjects.ts` file.

Reply:
ok, makes sense for me. Do you want me to start it on this ticket? 👀

Reply:
Yes, but only for the validations you added. To move them to the `app/routes/projects/archive.tsx` action.